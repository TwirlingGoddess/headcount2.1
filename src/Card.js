import React from 'react';
import PropTypes from 'prop-types'

const Card = ({index, location, stats, selectDistrict}) => {

	const displayData =
		Object.keys(stats).map((date, index) => {
		if(stats[date] > .5){
			return(
				<li key={index} className="purple">
					{date}: {stats[date]}
				</li>
			)
		} else {
			return(
				<li key={index} className="teal">
					{date}: {stats[date]}
				</li>
			)
		}
	})


	return(
		<div className="Card" onClick={selectDistrict}>
			<h2>{location}</h2>
			<ul>{displayData}</ul>
		</div>
	)
}

Card.propTypes = {
  selectDistrict: PropTypes.func
}

Card.propTypes = {
  location: PropTypes.string
}

Card.propTypes = {
  stats: PropTypes.object
}

Card.propTypes = {
  index: PropTypes.number
}

export default Card