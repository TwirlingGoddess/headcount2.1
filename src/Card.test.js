import React from 'react';
import { shallow } from 'enzyme';
import Card from './Card';

describe('Card', () => {
	let wrapper
	let mockData
	let expectedId
	let selectMock
	let clickedCard

	beforeEach(() => {
		expectedId = 7
	  selectMock = jest.fn()
		mockData = { 2004: 0.75, 2005: 0.9 }
		wrapper = shallow(<Card location="Colorado" 
	    											stats={mockData}
														selectDistrict={selectMock}
														id={expectedId} 
	    								/>)
		clickedCard = wrapper.find('.Card')
	})

	it('should match the snapshot', () => {
		expect(wrapper).toMatchSnapshot()
	})

  it('should render each data point as an aside', () => { 
    expect(wrapper.find('li').length).toEqual(2)
  })

	it('should invoke the toggleCard function when a card is clicked', () => {
    clickedCard.simulate('click')
    expect(selectMock).toBeCalled()
  })

  it('should invoke the toggleCard function with the correct id', () => {
    clickedCard.simulate('click')
    expect(selectMock).toHaveBeenCalledWith(expectedId)
  })
})
