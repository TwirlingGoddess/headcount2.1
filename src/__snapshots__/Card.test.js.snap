// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    id={7}
    location="Colorado"
    selectDistrict={[Function]}
    stats={
        Object {
            "2004": 0.75,
            "2005": 0.9,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Colorado
</h2>,
        <ul>
          <li
                    className="purple"
          >
                    2004
                    : 
                    0.75
          </li>
          <li
                    className="purple"
          >
                    2005
                    : 
                    0.9
          </li>
</ul>,
      ],
      "className": "Card",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Colorado",
        },
        "ref": null,
        "rendered": "Colorado",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="purple"
>
              2004
              : 
              0.75
</li>,
            <li
              className="purple"
>
              2005
              : 
              0.9
</li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "2004",
                ": ",
                0.75,
              ],
              "className": "purple",
            },
            "ref": null,
            "rendered": Array [
              "2004",
              ": ",
              0.75,
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "2005",
                ": ",
                0.9,
              ],
              "className": "purple",
            },
            "ref": null,
            "rendered": Array [
              "2005",
              ": ",
              0.9,
            ],
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Colorado
</h2>,
          <ul>
            <li
                        className="purple"
            >
                        2004
                        : 
                        0.75
            </li>
            <li
                        className="purple"
            >
                        2005
                        : 
                        0.9
            </li>
</ul>,
        ],
        "className": "Card",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Colorado",
          },
          "ref": null,
          "rendered": "Colorado",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className="purple"
>
                2004
                : 
                0.75
</li>,
              <li
                className="purple"
>
                2005
                : 
                0.9
</li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "2004",
                  ": ",
                  0.75,
                ],
                "className": "purple",
              },
              "ref": null,
              "rendered": Array [
                "2004",
                ": ",
                0.75,
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "2005",
                  ": ",
                  0.9,
                ],
                "className": "purple",
              },
              "ref": null,
              "rendered": Array [
                "2005",
                ": ",
                0.9,
              ],
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
